lib/checkcondition.cpp

```diff
@@ -784,8 +784,9 @@ static inline T getvalue(const int test, const T value1, const T value2)
         return value2;
     case 5:
         return std::numeric_limits<T>::max();
+    default: 
+      return 0;
     };
-    return 0;
```

Comment:
In general I like stylistically that there is a return/throw at the end of a function that returns non-void. I want it to be easy to see that there is no way for the function to return without a value.

Therefore, I general I would dislike that you move this return into the switch.

In this specific case I am not that much against it because it is relatively easy to see anyway. but for consistency, I would like to keep the return where it is.